### PR TITLE
Fixed issue 231

### PR DESCRIPTION
**Pull request by MSattrtand**
Fixes issue #231.

image
If an incorrect query causes an exception, the query is logged now.

### CODE REVIEW

**Comment by blcham** [comment]
1) for referencing issues use # ... as it is very well integrated with github  + use more specific commits:
     Fixed issues 231 ---> [#231] Improve logging for queries when parse error occurs

**Comment by MSattrtand** [comment]
The stack trace shows that the exception has occurred at the perform function of the ParserARQ class right here:
image
The last of the s-pipes functions that may cause the exception are ApplyConstructModule and BindBySelectModule, so if we assume that foreign libraries are working correctly, the main reason the parser may fail is that our query is truly incorrect. Therefore, I thought logging only the query and the stack trace was enough for debugging.
NB: There are more functions that may (theoretically) send incorrect queries to the parser; ApplyConstructModule and BindBySelectModule are the only modules I can properly test because they are called during the execution of the Hello, World script.

**Comment by blcham** [comment]
> The stack trace shows that the exception has occurred at the perform function of the Parse .....

Not sure what you mean, but I am saying that the error you are logging is not good enough because you won't be able to find the error within the query. Try to read what you sent me above and find the error based on the coordinates: in you output:
`line:32 column 7`

File: s-pipes-core/src/main/java/cz/cvut/spipes/modules/ApplyConstructModule.java [hunk 1]
```diff
@@ -111,9 +111,9 @@ public ExecutionContext executeSelf() {
         Model inferredModel = ModelFactory.createDefaultModel();
 
         List<Construct> constructQueriesSorted = constructQueries
-            .stream().map(r -> r.as(Construct.class))
```
Comment (blcham) [inline s-pipes-core/src/main/java/cz/cvut/spipes/modules/ApplyConstructModule.java:114]: I do not understand why you get so many formatting changes. We should investigate what the difference is in our setting (maybe use some kind of prettifier of the code to enforce the same formatting).

Anyway, if having too many formatting changes, do the formatting in a separate commits so the actual change is readable.

File: s-pipes-core/src/main/java/cz/cvut/spipes/modules/BindBySelectModule.java [hunk 2]
```diff
@@ -25,33 +26,42 @@ public class BindBySelectModule extends AbstractModule  {
...
-        return ExecutionContextFactory.createContext(
-            this.createOutputModel(isReplace, ModelFactory.createDefaultModel()),
-            variablesBinding
-        );
+            return ExecutionContextFactory.createContext(
+                    this.createOutputModel(isReplace, ModelFactory.createDefaultModel()),
+                    variablesBinding
+            );
+        }
+        catch (Exception e) {
```
Comment (blcham) [inline s-pipes-core/src/main/java/cz/cvut/spipes/modules/BindBySelectModule.java:58]: you are duplicating this part of code with the one above ... we need to use some utility function

File: s-pipes-core/src/main/java/cz/cvut/spipes/modules/ApplyConstructModule.java [hunk 3]
```diff
@@ -127,29 +127,38 @@ public ExecutionContext executeSelf() {
...
-                    LOG.debug("... saving module's partially computed output to file {}.", saveModelToTemporaryFile(constructedModel));
-                }
+                    if (AuditConfig.isEnabled() || ExecutionConfig.getEnvironment().equals(Environment.development)) {
+                        LOG.debug("... saving module's partially computed output to file {}.", saveModelToTemporaryFile(constructedModel));
+                    }
 
-                inferredInSingleIterationModel = ModelFactory.createUnion(inferredInSingleIterationModel, constructedModel);
+                    inferredInSingleIterationModel = ModelFactory.createUnion(inferredInSingleIterationModel, constructedModel);
+                }
+                catch (Exception e) {
```
Comment (blcham) [inline s-pipes-core/src/main/java/cz/cvut/spipes/modules/ApplyConstructModule.java:155]: we are trying to describe QueryParseException not generic exception.

**Review by blcham — CHANGES_REQUESTED** [review]
@MSattrtand I created a new PR -- https://github.com/kbss-cvut/s-pipes/pull/241.

Also see my comments here -- https://github.com/kbss-cvut/s-pipes/issues/231#issuecomment-2210212910

Read my comments, close this PR, and continue work on the newly provided PR (make new PRs to merge to this branch)

**Comment by blcham** [comment]
@MSattrtand see my comments and https://github.com/kbss-cvut/s-pipes/pull/241

**Pull request closed without merging**